src/commands.ts

```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
 		project = args.fsPath;
 	}
 
-	await projectHelper.buildProjects([project], commandObserver).then(async result => {
-		if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
-			var filePath = await projectHelper.getOutputFilePath(project);
-			if (filePath && fs.existsSync(filePath)) {
-				var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
-				if (connection) {
-					vscode.workspace.openTextDocument(filePath).then(doc => {
-						vscode.window.showTextDocument(doc, vscode.ViewColumn.Active, false).then(() => {
-							let filePath = doc.uri.toString();
-							azdata.queryeditor.connect(filePath, connection.connectionId).then(() => azdata.queryeditor.runQuery(filePath, undefined, false));
+	try {
+		await projectHelper.buildProjects([project], commandObserver).then(async result => {
+			if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
+				await projectHelper.setOutputFilePath(project, commandObserver);
+				var filePath = commandObserver.outputFilePath;
+				if (filePath && fs.existsSync(filePath)) {
+					var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
+					if (connection) {
+						let projectFileText = fs.readFileSync(filePath, 'utf8');
+						azdata.queryeditor.connect(filePath, connection.connectionId).then(() => {
+							commandObserver.logToOutputChannel(localize('extension.deploymentStartedMessage', '\nDeployment started {0}.', new Date().toLocaleString()));
+							commandObserver._client.sendRequest("query/executeDeployString", {owner_uri: filePath, query:projectFileText})
```

Comment:
I'd probably call the command ```deployScript``` or something.

Reply:
Or maybe ```deployString``` to be consistent, e.g. we change ```execute``` to ```deploy```...

Reply:
Hmmmmm, I added the command to execute_request in pgtoolsservice, so I was trying to follow the pattern.

Reply:
Probably rename it there too? I agree we should be consistent...
I guess what confuses me now is ```executeString``` should be ```executeExecuteString``` if command itself is called "execute string". With current naming convention, if command is called "deploy string" then it should say just that, me thinks.

Reply:
Hm... although looking on the other side they call them ```'EXECUTE_DOCUMENT_SELECTION_REQUEST', 'EXECUTE_STRING_REQUEST'```, which appear sot be that "execute request" is a nomenclature they use. That makes just "string" a request name, which is weird.
With this, maybe we can call it ```executeDeploy``` and on the python side call it ```EXECUTE_DEPLOY_REQUEST```? Would that make sense?

Reply:
Yeah that works. More consistent with what they have and still makes sense.